lisp/mastodon-http.el

```diff
@@ -82,21 +83,31 @@ Authorization header is included by default unless UNAUTHENTICED-P is non-nil."
                       "&")))
         (url-request-extra-headers
 	 (append
-	  (unless unauthenticed-p
+	  (unless (and unauthenticed-p
+                       (string= "oauth2" mastodon-auth-mechanism))
 	    `(("Authorization" . ,(concat "Bearer " (mastodon-auth--access-token)))))
 	  headers)))
     (with-temp-buffer
-      (url-retrieve-synchronously url))))
+      (if (string= "oauth2" mastodon-auth-mechanism)
```

Comment:
As mentioned in other comment: Don't use `oauth2-url-retrieve-synchronously` when making an unauthenticated call.

Reply:
Instead of this, I am going with the modification in `mastodon-client--register`. This will make sure the settings are changed only when needed and will keep logical clarity intact.

In my point of view, we don't want to use oauth2 to register the client. Which has nothing to do with generic post requests directly. Although it is true that not making calls to `oauth2-url-retrieve-synchronously` is a solution, it is at a place which is too broad.

I hope I am making the reasoning clear.

Reply:
Now that I'm thinking about it, fixing it at the POST call will prevent future bugs too.

Reply:
@hdurer this is regarding the fix being put at two places.